bin/check_practice_exercise_order.exs

```diff
@@ -0,0 +1,52 @@
+Mix.install([
+  {:jason, "~> 1.2"}
+])
+
+{opts, _} =
+  OptionParser.parse!(
+    System.argv(),
+    strict: [write: :boolean],
+    aliases: [w: :write]
+  )
+
+config =
+  "./config.json"
+  |> File.read!()
+  |> Jason.decode!()
+
+sort_by = fn exercise ->
+  {exercise["difficulty"], exercise["name"]}
+end
+
+reordered_config =
+  put_in(
+    config,
+    ["exercises", "practice"],
+    Enum.sort_by(config["exercises"]["practice"], sort_by)
+  )
+
+desired_exercise_order =
+  reordered_config
+  |> get_in(["exercises", "practice"])
+  |> Enum.map(& &1["slug"])
+
+actual_exercise_order =
+  config
+  |> get_in(["exercises", "practice"])
+  |> Enum.map(& &1["slug"])
+
+if opts[:write] do
+  {reordered_file, 0} =
+    System.cmd("jq", [".exercises.practice|=sort_by(.difficulty, .name)", "config.json"])
```

Comment:
This sorting logic will starting failing us once there is a difficulty-1 exercise that comes alphabetically before hello-world 😞 not sure what to do about that.

Reply:
Why does `hello-world` need to come first? 
We Ould add out own optional field `"hardcoded_order" : 1` to something.

Reply:
`hello-world` is a special exercise in that it isn't allowed to have any prerequisites and it has the purpose of making the student run the simplest test suite and do the simplest possible code change in a new language. Every other exercise, including concept exercises, are all blocked until you do `hello-world`.

A custom key on that one exercise should work 🤔 currently there are no linter rules disallowing it.

Reply:
I managed to come up with some `jq` comments that work with a custom `"order"` field that is optional on an exercise and can be used to override the default alphabetical order. This way we will be able to control order of other exercises too, if that ever matters within the same difficulty group.